crates/warg/src/hash.rs

```diff
@@ -0,0 +1,105 @@
+use std::{fmt, ops::Deref, str::FromStr};
+
+use base64::{decode, encode};
+use digest::Digest;
+use thiserror::Error;
+
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub enum HashAlgorithm {
```

Comment:
Usually a good idea for `pub enum`s:

```suggestion
#[non_exhaustive]
pub enum HashAlgorithm {
```